vkcs/data_source_vkcs_publicdns_zone.go

```diff
@@ -0,0 +1,103 @@
+package vkcs
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourcePublicDNSZone() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourcePublicDNSZoneRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"id": {
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
id should not be required. We use data source to find an object

Reply:
Public API needs to provide filtering parameters to list zones route, so it can be found using filters. Without this, I believe that it's not supposed to retrieve information about a specific zone using other arguments except it's ID.

Reply:
it does